slither/__main__.py

```diff
@@ -166,7 +165,14 @@ def get_detectors_and_printers() -> Tuple[
     printers = [p for p in printers_ if inspect.isclass(p) and issubclass(p, AbstractPrinter)]
 
     # Handle plugins!
-    for entry_point in iter_entry_points(group="slither_analyzer.plugin", name=None):
+    if sys.version_info >= (3, 10):
+        entry_points = metadata.entry_points(group="slither_analyzer.plugin")
```

Comment:
It may be possible to use this implementation in 3.8+ as well if we added a dependency on `importlib_metadata >= 3.6` and imported `metadata` through it.